Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/ComponentEditor.cpp

```diff
@@ -150,7 +151,15 @@ namespace AzToolsFramework
         if (DocumentPropertyEditor::ShouldReplaceRPE())
         {
             // Instantiate the DPE without the RPE
-            m_adapter = AZStd::make_shared<AZ::DocumentPropertyEditor::ComponentAdapter>();
+            if (Prefab::IsInspectorOverrideManagementEnabled())
+            {
+                m_adapter = AZStd::make_shared<Prefab::PrefabComponentAdapter>();
+            }
+            else
+            {
+                m_adapter = AZStd::make_shared<AZ::DocumentPropertyEditor::ComponentAdapter>();
```

Comment:
is this all we need to ensure that opening UI editor with dpe enabling doesn't crash?

Reply:
@srikappa-amzn I am working on that part in a separate PR.